app/assets/javascripts/mpa/ontology/taxa/NCBITaxonomy.ts

```diff
@@ -0,0 +1,71 @@
+import {Ontology} from '../Ontology'
+import {NCBITaxon} from './NCBITaxon';
+import {postJSON} from "../../../utils";
+
+type OntologyId = number;
+
+const TAXA_BATCH_SIZE = 100
+const LINEAGE_BATCH_SIZE = 100
+
+const TAXA_URL = "/private_api/taxa"
+const LINEAGE_URL = "/private_api/lineages"
+
+export class NCBITaxonomy extends Ontology<OntologyId, NCBITaxon>
+{
+    async getTaxaInfo(ids: OntologyId[]){
+        ids = ids.filter(id => 
+            !this._definitions.has(id) 
+            ||  !this._definitions.get(id).hasOwnProperty('name'))
+
+        // get taxa info
+        for (let i = 0; i < ids.length; i += TAXA_BATCH_SIZE) 
+        {
+            const data = JSON.stringify({
+                taxids: ids.slice(i, i + TAXA_BATCH_SIZE)
+            });
+
+            const res = await postJSON(TAXA_URL, data);
```

Comment:
Reading data from the Unipept API should be handled by the webworker. this webworker should keep a cache of all taxa-data that was requested before.